### PR TITLE
Marketplace: fix sideways scrolling of the my subscriptions table on Chromium based browsers

### PR DESCRIPTION
**Pull request by raicem**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
We used `subgrids` to align the columns of tables on the my subscriptions page (#41505). This worked great except it prevented sideways scrolling of the tables on Chromium browsers. It worked fine on Firefox. 

This PR replaces the use of `subgrid` with setting a fixed width for the product name column. 

We only set a fixed width for that column because it's the only one that can vary in width dramatically, depending on the product name and a badge. 

### How to test the changes in this Pull Request:

1. Checkout this branch and build the assets
2. Load my subscriptions page on a narrow viewport with a Chromium based browser (Google Chrome, Brave, Vivaldi etc...) Confirm you can sideways scroll the table.
3. 
4. Load my subscriptions page on a narrow viewport with Firefox. Confirm you can sideways scroll the table.
5. Please test around this issue. 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss [hunk 1]
```diff
@@ -242,7 +242,7 @@
 	padding: 6px 12px;
 }
 .woocommerce-marketplace__my-subscriptions
-	.components-button.is-secondary:hover:not(:disabled) {
```
Comment (raicem) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss]: I think my linter fixed this automatically, not sure what's this about.

**Review by andfinally — COMMENTED** [review]
I'm finding that the `is-scrollable-left` and `is-scrollable-right` classes are now being added to `woocommerce-marketplace__my-subscriptions__table` on Chrome where they weren't with subgrid. On both Chrome and FF, those classes are remaining on the table, and the side fade is still shown even when the table is at full width.

**Review by andfinally — APPROVED** [review]
I think this should do the trick, thanks for jumping in to fix this @raicem!

After looking more carefully, I think the overflow fade issue is a bug with the WooCommerce Table component. When you load the page at a certain width, the table renders as expected. It's only when you resize your viewport in dev tools that you get the unexpected fade. The component probably needs a viewport resize handler.

-------

EDIT

Interesting, it does have a [resize listener](https://github.com/woocommerce/woocommerce/blob/0d67a6aaf1293ab0d0315de7a4d203e22e71e931/packages/js/components/src/table/table.tsx#L166-L182). Maybe it's not working properly.

**Comment by raicem** [comment]
> I'm finding that the is-scrollable-left and is-scrollable-right classes are now being added to woocommerce-marketplace__my-subscriptions__table on Chrome where they weren't with subgrid. 

That's a valuable observation! So maybe it's not just a CSS subgrid issue but maybe a combination with JavaScript as well. This is getting tricky! Maybe we'll merge this as is anyways.

**Review by mlaetitia — APPROVED** [review]
Tested with Chrome, Arc and Firefox - worked like a charm in all

This does make the columns have different sizes again - what you tried to fix with subgrid. Assuming that is ok and that it's more important to have the scroll than the columns sizes matching (which I agree) - I'm approving this one.

**Review by Dan-Q — APPROVED** [review]
Tested with Firefox, Safari, and Edge. Breaks my heart to lose the power of the subgrid, but this is solid. 👍

**Pull request merged**